pkg/sql/catalog/table_elements.go

```diff
@@ -72,9 +72,9 @@ type Mutation interface {
 	// nil otherwise.
 	AsIndex() Index
 
-	// AsConstraint returns the corresponding Constraint if the mutation
-	// is on a constraint, nil otherwise.
-	AsConstraint() Constraint
+	// AsConstraintWithoutIndex returns the corresponding WithoutIndexConstraint
+	// if the mutation is on a constraint not backed by an index, nil otherwise.
+	AsConstraintWithoutIndex() WithoutIndexConstraint
```

Comment:
Good, The change in name is really needed! (I did similar things myself when hacking, and I call them `NonIndexBackedConstraint`)